llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1629,6 +1634,11 @@ static bool eliminateConstraints(Function &F, DominatorTree &DT, LoopInfo &LI,
 
     ICmpInst::Predicate Pred;
     if (!CB.isConditionFact()) {
+      if (Value *X; match(CB.Inst, m_Intrinsic<Intrinsic::abs>(m_Value(X)))) {
+        AddFact(CmpInst::ICMP_SGE, CB.Inst, X);
+        continue;
+      }
+
```

Comment:
I think all non-conditional facts should be handled here, so could we add `llvm_unreachable()` below and remove the extra check `    if (CB.isConditionFact()) {` as a follow-up?

Reply:
yeah